code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -5,7 +5,81 @@
 		return null
 	return ..()
 
+/mob/living/carbon/hitby(atom/movable/AM as mob|obj,var/speed = THROWFORCE_SPEED_DIVISOR)
+	..(AM, speed)
+	var/t_him = "it"
+	if (src.gender == MALE)
+		t_him = "him"
+	else if (src.gender == FEMALE)
+		t_him = "her"
+	var/show_ssd
+	var/mob/living/carbon/human/H = src
+	if(istype(H)) show_ssd = H.species.show_ssd
+	if(show_ssd && !client && !teleop)
```

Comment:
Same comments as above. `istype(src)` taboo and multiple uses of an unguarded `H` variable.